src/balancer/OceanPool.ts

```diff
@@ -800,15 +810,18 @@ export class OceanPool extends Pool {
       this.logger.error('ERROR: Too much reserve to add')
       return null
     }
-    const txid = await super.approve(
-      account,
-      this.oceanAddress,
-      poolAddress,
-      this.web3.utils.toWei(amount)
-    )
-    if (!txid) {
-      this.logger.error('ERROR: OCEAN approve failed')
-      return null
+    const oceanAllowance = await this.allowance(this.oceanAddress, account, poolAddress)
+    if (new Decimal(oceanAllowance).lt(amount)) {
```

Comment:
Not sure if we need to make this check here as well, isn't the check inside the approve method enough? Now when there are not enough tokens approved we double check

Reply:
The need of checking twice came up because the approve method returns `null` if the allowance is already bigger that the amount needed, and in the upper methods we if we call approve we are expecting a `txid` so we can proceed with the logic. The approve method also returns a null response in case the approve fails and we check if the txid is received from the approve method. So we check the allowance in the upper methods also and call approve only when necessary. In my opinion we can remove the check in approve method or we can return something different than null if the allowance is bigger than the amount.